### PR TITLE
Wip/lapi cscli options

### PR DESCRIPTION
**Pull request by buixor**
- makes the options more uniform between `alerts` and `decisions`
 - more fool-proof opt checks
 - reflect swagger changes to api filters

### CODE REVIEW

File: cmd/crowdsec-cli/bouncers.go [hunk 1]
```diff
@@ -130,20 +128,20 @@ To list/add/delete bouncers
...
 	cmdBouncers.AddCommand(cmdBouncersAdd)
 
 	var cmdBouncersDelete = &cobra.Command{
 		Use:     "delete",
 		Short:   "delete bouncer",
-		Long:    `delete bouncer`,
-		Example: `cscli bouncers delete --name test [--ip 1.2.3.4]`,
-		Args:    cobra.MaximumNArgs(1),
+		Long:    `delete bouncer with the given name`,
+		Example: `cscli bouncers delete MyTestBouncer`,
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/bouncers.go:139]: small fix, but can you use the same name for all bouncer in example ? upper you used  `MyBouncerName`
maybe you can just write : `<bouncer_name>` ?

File: cmd/crowdsec-cli/decisions.go [hunk 2]
```diff
@@ -156,50 +173,32 @@ cscli decisions list --range 1.2.3.4/32
 
 			alerts, _, err := Client.Alerts.List(context.Background(), filter)
 			if err != nil {
+				log.Printf("filter was : %s", spew.Sdump(filter))
+				params, _ := qs.Values(filter)
+				log.Printf("filter as : %s", params.Encode())
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/decisions.go:178]: can you remove the debug please ?

File: cmd/crowdsec-cli/decisions.go [hunk 3]
```diff
@@ -156,50 +173,32 @@ cscli decisions list --range 1.2.3.4/32
 
 			alerts, _, err := Client.Alerts.List(context.Background(), filter)
 			if err != nil {
+				log.Printf("filter was : %s", spew.Sdump(filter))
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/decisions.go:176]: can you remove the debug please ?

File: cmd/crowdsec-cli/decisions.go [hunk 4]
```diff
@@ -156,50 +173,32 @@ cscli decisions list --range 1.2.3.4/32
...
+				params, _ := qs.Values(filter)
+				log.Printf("filter as : %s", params.Encode())
 				log.Fatalf("Unable to list decisions : %v", err.Error())
 			}
 
 			err = DecisionsToTable(alerts)
 			if err != nil {
+				log.Printf("filter was : %s", spew.Sdump(filter))
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/decisions.go:184]: same here please

File: cmd/crowdsec-cli/decisions.go [hunk 5]
```diff
@@ -269,22 +267,17 @@ Args :
 			log.Info("Decision successfully added")
 		},
 	}
+	cmdDecisionsAdd.Flags().StringVarP(&Duration, "duration", "d", "4h", "Decision duration (ie. 1h,4h,30m)")
+	cmdDecisionsAdd.Flags().StringVarP(&Scope, "scope", "s", types.Ip, "Decision scope (ie. ip,range,username)")
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/decisions.go:271]: isn't this flag redundant with the main flag of `cmdDecisions` ?

File: cmd/crowdsec-cli/decisions.go [hunk 6]
```diff
@@ -269,22 +267,17 @@ Args :
...
+	cmdDecisionsAdd.Flags().StringVarP(&Reason, "reason", "R", "", "Decision reason (ie. spam)")
+	cmdDecisionsAdd.Flags().StringVarP(&Type, "type", "t", "ban", "Decision type (ie. ban,captcha,throttle)")
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/decisions.go:273]: isn't this flag redundant with the main flag of `cmdDecisions` ?

File: cmd/crowdsec-cli/utils.go [hunk 7]
```diff
@@ -39,6 +40,36 @@ func indexOf(s string, slice []string) int {
 	return -1
 }
 
+func manageCliDecisionAlerts(ip *string, ipRange *string, scope *string, value *string) error {
+	log.Printf("checking stuff : ip:%s range:%s scope:%s value:%s", *ip, *ipRange, *scope, *value)
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/utils.go:44]: can you remove this message please ? ou rewrite it and put it in debug ?

File: pkg/models/localapi_swagger.yaml [hunk 8]
```diff
@@ -355,11 +355,11 @@ paths:
           required: false
           type: boolean
           description: 'only return alerts with decisions not expired yet'
-        - name: alert_source
-          in: query
-          required: false
-          type: string
-          description: only return alerts with matching source (ie. cscli/crowdsec)
+        # - name: alert_source
```
Comment (AlteredCoder) [inline pkg/models/localapi_swagger.yaml:358]: can we remove this block of comment please?

File: cmd/crowdsec-cli/utils.go [hunk 9]
```diff
@@ -39,6 +40,36 @@ func indexOf(s string, slice []string) int {
...
+	switch strings.ToLower(*scope) {
+	case "ip":
+		*scope = types.Ip
+		if *ip != "" {
+			*value = *ip
+		}
+		if *value != "" {
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/utils.go:51]: may the `value`  variable be empty ? if yes what happen ?

File: cmd/crowdsec-cli/utils.go [hunk 10]
```diff
@@ -39,6 +40,36 @@ func indexOf(s string, slice []string) int {
...
+			if ipRepr == nil {
+				return fmt.Errorf("%s isn't a valid ip", *value)
+			}
+		}
+	case "range":
+		*scope = types.Range
+		if *ipRange != "" {
+			*value = *ipRange
+		}
+		if *value != "" {
```
Comment (AlteredCoder) [inline cmd/crowdsec-cli/utils.go:62]: may the `value`  variable be empty ? if yes what happen ?

File: cmd/crowdsec-cli/bouncers.go [hunk 11]
```diff
@@ -97,11 +94,12 @@ To list/add/delete bouncers
 		Use:     "add",
 		Short:   "add bouncer",
 		Long:    `add bouncer`,
-		Example: `cscli bouncers add --name test [--ip 1.2.3.4]`,
-		Args:    cobra.MaximumNArgs(1),
+		Example: `cscli bouncers add MyBouncerName [--ip 1.2.3.4]`,
```
Comment (buixor) [inline cmd/crowdsec-cli/bouncers.go:97]: Can we remove the `--ip` here, as it is ignored by LAPI ?

File: cmd/crowdsec-cli/decisions.go (hunk 6 above)
Comment (buixor) [reply cmd/crowdsec-cli/decisions.go:273]: yes that,s the plan, to override the defaults

File: cmd/crowdsec-cli/utils.go (hunk 9 above)
Comment (buixor) [reply cmd/crowdsec-cli/utils.go:51]: yes as it's used for `list` command as well, it's `add` and `delete` commands that check that it's coherent

**Pull request merged**